app/src/main/java/com/mapbox/maps/testapp/examples/sky/SkyLayerSnapshotterActivity.kt

```diff
@@ -77,28 +83,17 @@ class SkyLayerSnapshotterActivity : AppCompatActivity() {
           .build()
       )
       setUri(Style.MAPBOX_STREETS)
-      start(object : Snapshotter.SnapshotReadyCallback {
-        override fun onStyleLoaded(style: Style) {
-          val skyLayer = SkyLayer("sky_snapshotter")
-          skyLayer.skyType(SkyType.ATMOSPHERE)
-          skyLayer.skyAtmosphereSun(listOf(0.0, 90.0))
-          style.addLayer(skyLayer)
-        }
-
-        override fun onSnapshotCreated(snapshot: Expected<MapSnapshotInterface?, String?>) {
-          if (snapshot.isError) {
-            Logger.e(TAG, "${snapshot.error}")
-          } else {
-            val image = snapshot.value!!.image()
-            val configBmp: Bitmap.Config = Bitmap.Config.ARGB_8888
-            val bitmap: Bitmap = Bitmap.createBitmap(image.width, image.height, configBmp)
-            val buffer: ByteBuffer = ByteBuffer.wrap(image.data)
-            bitmap.copyPixelsFromBuffer(buffer)
-            maneuverView.setImageBitmap(bitmap)
-            maneuverCaption.visibility = View.VISIBLE
-          }
+      start {
+        it?.let { snapshot ->
+          val image = snapshot.image()
+          val configBmp: Bitmap.Config = Bitmap.Config.ARGB_8888
+          val bitmap: Bitmap = Bitmap.createBitmap(image.width, image.height, configBmp)
+          val buffer: ByteBuffer = ByteBuffer.wrap(image.data)
+          bitmap.copyPixelsFromBuffer(buffer)
```

Comment:
can we expose extension functions to `Image` class to conveniently convert it to `Bitmap`?

Reply:
or extension function to MapSnapshotInterface, which allows to get the image as bitmap.

Reply:
Great idea!